nbgrader/apps/collectapp.py

```diff
@@ -0,0 +1,128 @@
+import os
+import glob
+import shutil
+from collections import defaultdict
+import dateutil
+import dateutil.parser
+
+from IPython.utils.traitlets import Unicode, List, Bool
+
+from nbgrader.apps.baseapp import TransferApp, transfer_aliases, transfer_flags
+from nbgrader.utils import check_mode
+
+
+aliases = {}
+aliases.update(transfer_aliases)
+aliases.update({
+})
+
+flags = {}
+flags.update(transfer_flags)
+flags.update({
+    'update': (
+        {'CollectApp' : {'update': True}},
+        "Update existing submissions with ones that have newer timestamps."
+    ),
+})
+
+def groupby(l, key=lambda x: x):
+    d = defaultdict(list)
+    for item in l:
+        d[key(item)].append(item)
+    return d
+
+class CollectApp(TransferApp):
+
+    name = u'nbgrader-collect'
+    description = u'Collect an assignment from the nbgrader exchange'
+
+    aliases = aliases
+    flags = flags
+
+    examples = """
+        Here we go...
+        """
+    
+    update = Bool(
+        False,
+        config=True,
+        help="Update existing submissions with ones that have newer timestamps."
+    )
+    
+    def init_args(self):
+        if len(self.extra_args) == 1:
+            self.assignment_id = self.extra_args[0]
+        else:
+            self.fail("Invalid number of argument, call as `nbgrader release ASSIGNMENT`.")
+
+    def _path_to_record(self, path):
+        filename = os.path.split(path)[1]
+        filename_list = filename.split('+')
+        if len(filename_list) != 3:
+            self.fail("Invalid filename: {}".format(filename))
+        username = filename_list[0]
+        timestamp = dateutil.parser.parse(filename_list[2])
```

Comment:
I added a new function in utils called `utils.parse_utc` that you probably want to use here. Otherwise you run into the risk of comparing timezone-aware timestamps with timezone-naive timestamps, etc., which is a mess. So `parse_utc` will just convert everything to UTC which makes comparison a lot easier later.